<launch>
  <!-- Configuration through the launch file -->
  <arg name="filename" default="$(find uuv_nc_parser)/config/test_data.nc" />
  <!-- Labels -->
  <arg name="x_label" default="xc"/>
  <arg name="y_label" default="yc"/>
  <arg name="z_label" default="zc"/>
  <arg name="time_label" default="time"/>
  <arg name="day_offset" default="0"/>
  <arg name="hour_offset" default="0"/>
  <arg name="minutes_offset" default="0"/>
  <!-- Current velocity labels -->
  <arg name="u_east_label" default="u_east"/>
  <arg name="v_north_label" default="v_north"/>
  <!-- Wind velocity labels -->
  <arg name="w_east_label" default="w_east"/>
  <arg name="w_north_label" default="w_north"/>


  <arg name="max_size_pc" default="300"/>
  <node pkg="uuv_nc_parser"
    type="connect_ocean_data.py"
    name="connect_ocean_data"
    output="screen">
    <rosparam subst_value="true">
      update_rate: 1
      filename: $(arg filename)
      max_size_pc: $(arg max_size_pc)
      labels:
        x: $(arg x_label)
        y: $(arg y_label)
        z: $(arg z_label)
        time: $(arg time_label)
      time_offset:
        day: $(arg day_offset)
        hour: $(arg hour_offset)
        min: $(arg minutes_offset)
      center_x: True
      center_y: True
      loop_time: True
      current_velocity:
        u_east: $(arg u_east_label)
        v_north: $(arg v_north_label)
      wind_velocity:
        w_east: $(arg w_east_label)
        w_north: $(arg w_north_label)
      variables:
      - h2s
      - salinity
      - temperature
      - turbidity
    </rosparam>
  </node>

</launch>
